source/common/http/conn_pool_grid.cc

```diff
@@ -247,11 +250,7 @@ ConnectionPool::Cancellable* ConnectivityGrid::newStream(Http::ResponseDecoder&
     createNextPool();
   }
   PoolIterator pool = pools_.begin();
-  if (http3_status_tracker_.isHttp3Broken()) {
-    ENVOY_LOG(trace, "HTTP/3 is broken to host '{}', skipping.", describePool(**pool),
-              host_->hostname());
-    // Since HTTP/3 is broken, presumably both pools have already been created so this
-    // is just to be safe.
+  if (!shouldAttemptHttp3()) {
```

Comment:
Can you add a comment here? In particular: why do create the next pool only when h3 should not be attempted?

Reply:
Thanks, the comment helps